proto/float32.go

```diff
@@ -0,0 +1,37 @@
+package proto
+
+import (
+	"math"
+	"unsafe"
+)
+
+var float32Codec = codec{
+	wire:   fixed32,
+	size:   sizeOfFloat32,
+	encode: encodeFloat32,
+	decode: decodeFloat32,
+}
+
+func sizeOfFloat32(p unsafe.Pointer, flags flags) int {
+	if p != nil {
+		if v := *(*float32)(p); v != 0 || flags.has(wantzero) {
```

Comment:
Out of interest, how does protobuf encoding handle `inf`/`nan`/`-0`?

Reply:
I don't see anything in the spec on that.

Reply:
K, it's probably being pedantic but negative zero would be lost if `wantzero` is set. See https://play.golang.org/p/DOImpBUaAcY

Reply:
Good point 👍